cloudify_agent/installer/operations.py

```diff
@@ -81,33 +88,66 @@ def configure(cloudify_agent, installer, **_):
 
 
 @operation
-@create_agent_config_and_installer
+@create_agent_config_and_installer()
 def start(cloudify_agent, **_):
     """
     Only called in "init_script"/"plugin" mode, where the agent is started
     externally (e.g. userdata script), and all we have to do is wait for it
     """
+    agent_name = cloudify_agent['queue']
     update_agent_record(cloudify_agent, AgentState.STARTING)
     tenant = cloudify_utils.get_tenant()
     client = get_client(
         amqp_user=tenant['rabbitmq_username'],
         amqp_pass=tenant['rabbitmq_password'],
         amqp_vhost=tenant['rabbitmq_vhost']
     )
-    agent_alive = utils.is_agent_alive(cloudify_agent['queue'], client)
 
-    if not agent_alive:
-        if ctx.operation.retry_number > 3:
-            ctx.logger.warning('Waiting too long for Agent to start')
-            update_agent_record(cloudify_agent, AgentState.NONRESPONSIVE)
-        return ctx.operation.retry(
-            message='Waiting for Agent to start...')
+    marked_nonresponsive = False
+    # we'll wait up to an hour for the agent to start. It really should
+    # start sooner than that, but we have to give up eventually.
+    start_time = time.time()
+    deadline = start_time + 3600
+    # we can't poll too often, because we need to give the agent a chance to
+    # respond. 10 seconds should be PLENTY even on very latency-deficient
```

Comment:
I'll just leave this here... https://web.archive.org/web/20220330010954/https://www.blug.linux.no/rfc1149/writeup/

But yeah, probably we can go a bit less pathological!

Reply:
used to be 15 before! I think 10 is also nice.